geomesa-memory/src/test/scala/org/locationtech/geomesa/memory/cqengine/utils/SampleFeatures.scala

```diff
@@ -0,0 +1,264 @@
+    "WhatFloat:Float:cq-index=navigable",
+    "WhatDouble:Double:cq-index=navigable",
+    "WhatBool:Boolean",
+    "When:Date:cq-index=navigable",
+    "*Where:Point:srid=4326",
+    "Why:String" // Why can have nulls
+  ).mkString(",")
+  val sftWithIndexes = SimpleFeatureTypes.createType("test2", specIndexes)
+
+  val cq = SFTAttributes(sft)
+  val ff = CommonFactoryFinder.getFilterFactory2
+
+  val MIN_DATE = new DateTime(2014, 1, 1, 0, 0, 0, DateTimeZone.forID("UTC"))
+  val seconds_per_year = 365L * 24L * 60L * 60L
+  val string = "foo"
+
+  def randDate = MIN_DATE.plusSeconds(scala.math.round(scala.util.Random.nextFloat * seconds_per_year)).toDate
+
+  val builder = new SimpleFeatureBuilder(sft)
+
+  val names = Array("Addams", "Bierce", "Clemens", "Damon", "Evan", "Fred", "Goliath", "Harry")
+
+  def getName: String = names(Random.nextInt(names.length))
+
+  /* get non-uniformly distributed booleans */
+  def randomBoolean(fracTrue: Double): java.lang.Boolean = {
+    if (Random.nextDouble() < fracTrue)
+      true
+    else
+      false
+  }
+
+  def getPoint: Point = {
+    val minx = -180
+    val miny = -90
+    val dx = 360
+    val dy = 180
+
+    val x = minx + Random.nextDouble * dx
+    val y = miny + Random.nextDouble * dy
+    WKTUtils.read(s"POINT($x $y)").asInstanceOf[Point]
+  }
+
+  def buildFeature(i: Int): SimpleFeature = {
+    builder.set("Who", getName)
+    builder.set("What", Random.nextInt(10))
+    builder.set("WhatLong", Random.nextInt(10).toLong)
+    builder.set("WhatFloat", Random.nextFloat * 10.0F)
+    builder.set("WhatDouble", Random.nextDouble() * 10.0)
+    builder.set("WhatBool", randomBoolean(0.4))
+    builder.set("When", randDate)
+    builder.set("Where", getPoint)
+    if (randomBoolean(0.4)) {
+      // make sure n_{null} != n_{not null}
+      builder.set("Why", string)
+    }
+    builder.buildFeature(i.toString)
+  }
+}
+
+/**
+  * List of sample filters to test sample features.
+  * Each is designed so they are likely to return non-zero results over a reasonably
+  * large (>= 1000) set of random features generated with SampleFeatures.buildFeature()
+  */
+object SampleFilters {
+  implicit def stringToFilter(s: String): Filter = ECQL.toFilter(s)
+  val ff = CommonFactoryFinder.getFilterFactory2
+
+  // big enough so there are likely to be points in them
+  val bbox1 = "POLYGON((-89 89, -1 89, -1 -89, -89 -89, -89 89))"
+  val bbox2 = "POLYGON((-180 90, 0 90, 0 0, -180 0, -180 90))"
+
+  val equalityFilters: Seq[Filter] = Seq(
+    "Who = 'Addams'",
+    "Who IN ('Addams', 'Bierce')",
+
+    "What = 5",
+    "What IN (1, 2, 3, 4)",
+    "What <> 5",
+
+    "WhatLong = 5",
+    "WhatLong <> 5",
+    "WhatLong IN (1, 2, 3, 4)",
+
+    "When = '2014-07-01T00:00:00.000Z'",
+    "When <> '2014-07-01T00:00:00.000Z'",
+
+    "WhatBool = true",
+    "WhatBool = false",
+
+    ff.notEqual(ff.property("What"), ff.literal(5))  // CQL parser doesn't like "What != 5"
```

Comment:
Surprisingly, yes. GeoTools parses ``What <> 5`` as

    Not(PropertyIsEqualTo(Property(What), Literal(5)))

This is:

    PropertyIsNotEqualTo(Property(What), Literal(5))